crates/cargo-contract/src/cmd/extrinsics/remove.rs

```diff
@@ -0,0 +1,147 @@
+    #[clap(long, value_parser = parse_code_hash)]
+    code_hash: Option<<DefaultConfig as Config>::Hash>,
+    #[clap(flatten)]
+    extrinsic_opts: ExtrinsicOpts,
+    /// Export the call output in JSON format.
+    #[clap(long, conflicts_with = "verbose")]
+    output_json: bool,
+}
+
+impl RemoveCommand {
+    pub fn is_json(&self) -> bool {
+        self.output_json
+    }
+
+    pub fn run(&self) -> Result<(), ErrorVariant> {
+        let artifacts = self.extrinsic_opts.contract_artifacts()?;
+        let transcoder = artifacts.contract_transcoder()?;
+        let signer = super::pair_signer(self.extrinsic_opts.signer()?);
+
+        let artifacts_path = artifacts.artifact_path().to_path_buf();
+
+        let final_code_hash = match (self.code_hash.as_ref(), artifacts.code.as_ref()) {
+            (Some(code_h), Some(_)) => {
+                Ok(code_h.0)
+            }
+            (Some(code_h), None) => {
+                Ok(code_h.0)
+            }
+            (None, Some(_)) => {
+                let interm_artifacts_code = artifacts.code_hash()?;
+                Ok(interm_artifacts_code)
+            }
+            (None, None) => {
+                Err(
+                    anyhow::anyhow!(
+                    "No code_hash was provided or contract code was not found from artifact file {}. Please provide a code hash with --code-hash argument or specify the path for artifacts files with --manifest-path", 
+                    artifacts_path.display()
+                ))
+            }
+        };
+
+        async_std::task::block_on(async {
+            let url = self.extrinsic_opts.url_to_string();
+            let client = OnlineClient::from_url(url.clone()).await?;
+            if let Some(code_removed) = self
+                .remove_code(
+                    &client,
+                    sp_core::H256(final_code_hash?),
+                    &signer,
+                    &transcoder,
+                )
+                .await?
+            {
+                let remove_result = code_removed.code_hash;
+
+                if self.output_json {
+                    println!("{}", &remove_result);
+                } else {
+                    name_value_println!("Code hash", format!("{remove_result:?}"));
+                }
+                Result::<(), ErrorVariant>::Ok(())
+            } else {
+                let art_code_hash = artifacts.code_hash();
+                let error_code_hash = hex::encode(art_code_hash?);
+                Err(anyhow::anyhow!(
+                    "This contract could not have been removed for the supplied code hash: {}", error_code_hash)
+                    .into())
+            }
+        })
+    }
+
+    async fn remove_code(
+        &self,
+        client: &Client,
+        code_hash: CodeHash,
+        signer: &PairSigner,
+        transcoder: &ContractMessageTranscoder,
+    ) -> Result<Option<api::contracts::events::CodeRemoved>, anyhow::Error> {
```

Comment:
`api::contracts` can be imported globally in a file since it's used more than once

Reply:
I have addressed this one: https://github.com/paritytech/cargo-contract/pull/837/commits/a857219036f9cc8aa933bdb9e49f43600787f659